shell/platform/android/io/flutter/embedding/android/FlutterFragment.java

```diff
@@ -1056,6 +1056,11 @@ public void onAttach(@NonNull Context context) {
     delegate.onAttach(context);
     if (getArguments().getBoolean(ARG_SHOULD_AUTOMATICALLY_HANDLE_ON_BACK_PRESSED, false)) {
       requireActivity().getOnBackPressedDispatcher().addCallback(this, onBackPressedCallback);
+      // By default, Android handles backs, and predictive back is enabled. This
+      // can be changed by calling setFrameworkHandlesBack. For example, the
+      // framework will call this automatically in a typical app when it has
+      // routes to pop.
```

Comment:
I'm going to push a commit to reword this comment to try to make it more clear.

ARG_SHOULD_AUTOMATICALLY_HANDLE_ON_BACK_PRESSED means that predictive back is enabled. onBackPressedCallback enabled means that Flutter is handling back gestures (typically because Flutter has Flutter routes to on its navigation stack). Disabled means that Android is handling back gestures, so it will either pop the Activity or go back to the phone's home screen.

So when the app starts, by default we let Android handle backs. If/when the Flutter app wants to change this, it calls setFrameworkHandlesBack and enables this callback.